lib/shopping_cart.rb

```diff
@@ -0,0 +1,54 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.delete('items').to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    detalles = {}
+    detalles[:name] = @name
+    detalles[:capacity] = @capacity
+    detalles
```

Comment:
You could also do something like the following: 

{name: @name, capacity: @capacity}

and have that be the only line in this method